llvm/lib/ProfileData/InstrProf.cpp

```diff
@@ -330,7 +303,17 @@ static std::optional<std::string> lookupPGONameFromMetadata(MDNode *MD) {
   return {};
 }
 
-// Returns the PGO object name. This function has some special handling
+// The PGO name has the format [<filepath>;]<function-name> where <filepath>; is
```

Comment:
nit: `s/<function-name>/<mangled-name>` since `getIRPGOObjectName` might be called on non-function objects (e.g., global variables).

Reply:
Where is `getIRPGOObjectName()` called on a global variable? My understanding is that it is only used on functions we want to instrument.

Reply:
> Where is getIRPGOObjectName() called on a global variable? 

It's only called on functions in top-of-the-tree. To add [type-profiling](https://discourse.llvm.org/t/rfc-dynamic-type-profiling-and-optimizations-in-llvm/74600/9) in instrFDO, a [pending pull request](https://github.com/llvm/llvm-project/pull/66825) will use it on virtual table definitions. This is the motivation for [pull request 70287](https://github.com/llvm/llvm-project/pull/70287) to refactor the function names and arguments (to take `GlobalObjects`).

Reply:
Oh I remember seeing this. Thanks for the pointer!